torchaudio/transforms.py

```diff
@@ -664,28 +664,41 @@ class Resample(torch.nn.Module):
         rolloff (float, optional): The roll-off frequency of the filter, as a fraction of the Nyquist.
             Lower values reduce anti-aliasing, but also reduce some of the highest frequencies. (Default: ``0.99``)
         beta (float or None): The shape parameter used for kaiser window.
+        dtype (torch.device, optional):
+            Determnines the precision that resampling kernel is pre-computed and cached. If not provided,
+            kernel is computed with ``torch.float64`` then cached as ``torch.float32``.
+            If you need higher precision, provide ``torch.float64``, and the pre-computed kernel is computed and
```

Comment:
A few questions here
- if the user provides `dtype=torch.float64`, should they still provide `.to(torch.float64)` afterwards or is that unnecessary?
- should we assume that it is preferable for the user to have the generation in `float64` even if they are using something like float16? using `.to(dtype)` in these cases will improve precision, but the resulting behavior will be different than the functional, which does the entire computation in the final dtype

Reply:
> * if the user provides `dtype=torch.float64`, should they still provide `.to(torch.float64)` afterwards or is that unnecessary?

It depends. If the transform object user is manipulating is only composed of `Resample`, then it makes no difference, but if the processing pipeline is composed of multiple transforms (say, resample -> spectrogram -> mel-scale), then user will have to call `.to` for the entire transforms. The result-wise it should be same.

> * should we assume that it is preferable for the user to have the generation in `float64` even if they are using something like float16? 

Yes, that's my view. The only reason we are adding `dtype` is otherwise we cannot provide a mean to resample in higher precision. We are not trying to get rid of the need for users to call `.to`. Therefore, in the doc, I only instructed the case where users absolutely are necessary to use it.

> using `.to(dtype)` in these cases will improve precision, but the resulting behavior will be different than the functional, which does the entire computation in the final dtype

I think 1. users can still provide `dtype=float16`, if they want the exact same result as `F.resample`. 2. it's okay to have small discrepancy here, as it presumably produces better result (the quality of resampled audio).